### ISSUE TITLE
Tutorial: how to use filters?

### ISSUE THREAD

**Issue by drandreaskrueger** [ethereum/web3.py#378]
* Version: 3.16.2
* Python: 3.5
* OS: linux

### What was wrong?
The example **filter** code just ended, without any output.

### How can it be fixed?

It took me ages to get your very first example working. I did not know what to do with the resulting `new_block_filter` object. 

I suspected I had to start some kind of eventloop, but I did not know how.  Finally I tried `.join()`, but the output was always nothing. Then suddenly for `.join(timeout=5)`, I got results, but only *AFTER* those 5 seconds, not already when the block appeared. 

Then, for this issue, when I replaced your `sys.stdout.write()` with `print()` ... it suddenly worked. What how why? It made me realize how to fix [your example](http://web3py.readthedocs.io/en/latest/filters.html#block-and-transaction-filters). Just add the \*grrr\* line:

```
import sys
from web3 import Web3, HTTPProvider, __version__
print (__version__) # or what is a better way?
web3 = Web3(HTTPProvider('http://localhost:8545'))

def blockfilter():
    
    def new_block_callback(block_hash):
        sys.stdout.write("New Block: {0}\n".format(block_hash))
        sys.stdout.flush() # without this NOTHING happened *grrr*
            
    new_block_filter = web3.eth.filter('latest')
    new_block_filter.watch(new_block_callback)
    
    return new_block_filter 

bf = blockfilter()

bf.join()  # mention at least this in your docs?
```

I suggest you give this example code somewhere on [filters.html](http://web3py.readthedocs.io/en/latest/filters.html), to show *how to use* such filter greenlets.

Thanks a lot. Very useful library your Web3.py. Keep up the good work.

**Comment by pipermerriam** [comment]
@djrtwo this might be useful for the work your doing on synchronous filters.

**Comment by drandreaskrueger** [comment]
Actually, this also works:

```
bf = blockfilter()
import time
time.sleep(10)
```
= doesn't need that call to `.join()`.

So the thing to make your [filter examples](http://web3py.readthedocs.io/en/latest/filters.html) working, is the `flush()` call.

**Comment by TelonChee** [comment]
Thanks for ur example! It's helpful to new beginners.

BTW:
Do i need remove a filter when finished parse data from contract pastEvents methods? If it does, how can i remove the filter, or just call filter's stopWatching method?

**Comment by dylanjw** [comment]
fixed in #631
